product/roundhouse/infrastructure.app/ApplicationConfiguraton.cs

```diff
@@ -124,10 +124,7 @@ public static void set_defaults_if_properties_are_not_set(ConfigurationPropertyH
             }
             if (string.IsNullOrEmpty(configuration_property_holder.EnvironmentNames))
             {
-                if (!string.IsNullOrEmpty(configuration_property_holder.EnvironmentName))
-                    configuration_property_holder.EnvironmentNames = configuration_property_holder.EnvironmentName;
-                else
-                    configuration_property_holder.EnvironmentNames = ApplicationParameters.default_environment_name;
+                configuration_property_holder.EnvironmentNames = ApplicationParameters.default_environment_name;

```

Comment:
This change is not quite correct. Still need to be able to pass the environment name from the CLI

Reply:
The configuration_property_holder is only set from environment "names" (in command line, where it's just a synonym). Is this something that needs to be available at the object for users of the RH library?